src/ngraph/op/fused/clamp.cpp

```diff
@@ -18,12 +18,105 @@
 #include "ngraph/builder/make_constant.hpp"
 #include "ngraph/op/maximum.hpp"
 #include "ngraph/op/minimum.hpp"
+#include "ngraph/runtime/reference/clamp.hpp"
+#include "ngraph/util.hpp"
 
 using namespace std;
 using namespace ngraph;
 
 constexpr NodeTypeInfo op::Clamp::type_info;
 
+namespace
+{
+    template <element::Type_t ET, typename T>
+    bool evaluate(const HostTensorPtr& arg, const HostTensorPtr& out, T min, T max, size_t count)
+    {
+        runtime::reference::clamp<T>(
+            arg->get_data_ptr<ET>(), out->get_data_ptr<ET>(), min, max, count);
+        return true;
+    }
+
+    bool evaluate_clamp(
+        const HostTensorPtr& arg, const HostTensorPtr& out, double min, double max, size_t count)
+    {
+        auto ceil_func = [](double x) { return std::ceil(x); };
+        auto floor_func = [](double x) { return std::floor(x); };
```

Comment:
Why ceil/floor? I have not seen a description of clamp that says you can't do something like 0.5 as min or max. Get rid of ceil/floor.